Hardcore.lua

```diff
@@ -552,6 +515,43 @@ function Hardcore:PLAYER_DEAD()
 	end
 end
 
+function Hardcore:PLAYER_TARGET_CHANGED()
+	if Hardcore_Character.grief_warning_conditions == GRIEF_WARNING_BOTH_FACTIONS then
+		if UnitIsPVP("target") and UnitGUID("target") ~= PLAYER_GUID then
+			local faction, _ = UnitFactionGroup("target")
+			if faction ~= nil then
+				if faction ~= PLAYER_FACTION then
+					local target_name, _ = UnitName("target")
+					Hardcore:ShowAlertFrame(ALERT_STYLES.hc_pvp_warning, "Target " .. target_name .. " is PvP enabled!")
+				elseif UnitPlayerControlled("target") then
+					local target_name, _ = UnitName("target")
+					Hardcore:ShowAlertFrame(ALERT_STYLES.hc_pvp_warning, "Target " .. target_name .. " is PvP enabled!")
+				end
+			end
+		end
+	elseif Hardcore_Character.grief_warning_conditions == GRIEF_WARNING_ENEMY_FACTION then
+		if UnitGUID("target") ~= PLAYER_GUID and UnitIsPVP("target")  then
+			local faction, _ = UnitFactionGroup("target")
+			if faction ~= nil then
+				if (faction ~= PLAYER_FACTION) then
+					local target_name, _ = UnitName("target")
+					Hardcore:ShowAlertFrame(ALERT_STYLES.hc_pvp_warning, "Target " .. target_name .. " is PvP enabled!")
+				end
+			end
+		end
+	elseif Hardcore_Character.grief_warning_conditions == GRIEF_WARNING_SAME_FACTION then
+		if UnitGUID("target") ~= PLAYER_GUID and UnitIsPVP("target")  then
+			local faction, _ = UnitFactionGroup("target")
+			if faction ~= nil then
+				if faction == PLAYER_FACTION and UnitPlayerControlled("target") then
+					local target_name, _ = UnitName("target")
+					Hardcore:ShowAlertFrame(ALERT_STYLES.hc_pvp_warning, "Target " .. target_name .. " is PvP enabled!")
+				end
+			end
```

Comment:
Same here, it usually creates cleaner-looking code to have less nesting imo. (Though this line gets longer)

```
if faction ~= nil and faction == PLAYER_FACTION and UnitPlayerControlled("target") then
```

Reply:
Maybe its just me but I'd rather keep the `faction ~= nil` explicit since it is an expected output of the unit faction.  condensing sounds good though

Reply:
Yeah, you're right its probably safe-type too. Loose typing is the wild west 🤠